tt_eager/queue/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ set(QUEUE_SRCS
 )
 
 add_library(queue OBJECT ${QUEUE_SRCS})
-target_link_libraries(queue PUBLIC metal_header_directories compiler_flags)
+target_link_libraries(queue PUBLIC metal_header_directories compiler_flags umd_device)
```

Comment:
Why do we have to link `umd_device` here? Curious because I don't think we needed to before so wondering what changed?

Reply:
You're right, it doesn't need to be linked here. Gonna push an update in a moment.

Reply:
Do we need them in the other tt_lib or ttnn targets as well?

Reply:
Yeah, everywhere I added `umd_device` is because I had the missing headers compiler error, coming from inside UMD.